src/main/java/com/zaxxer/hikari/proxy/ConnectionProxy.java

```diff
@@ -147,8 +147,9 @@ public final void markCommitStateDirty()
       return statement;
    }
 
-   private final void closeOpenStatements()
+   private final boolean closeOpenStatements()
    {
+      boolean success = true;
```

Comment:
Move the ``boolean success = true;`` inside the ``if (size > 0)`` conditional...